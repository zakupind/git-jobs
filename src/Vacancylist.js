import React from 'react';
import { connect } from 'react-redux';
import Vacancy from './Vacancy';
import Loader from './Loader';

class VacancyList extends React.Component {
    render() {
        return( 
            <div> {this.props.isFetching ? 
                <Loader /> :
                <div>
                    <span className="search__status">{this.props.statusSearch}</span>
                    <ul>
                        {this.props.jobs.map(job => (
                            <Vacancy key={job.id} {...job}/>
                        ))}
                    </ul>
                </div>
            }
            </div>
        )
    }
}

const mapStateToProps = state => ({
    jobs: state.search.jobs,
    statusSearch: state.search.statusSearch,
    isFetching: state.search.isFetching
});


export default connect(mapStateToProps, null)(VacancyList);