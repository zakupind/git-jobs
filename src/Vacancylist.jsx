import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Vacancy from './Vacancy';
import Loader from './Loader';

function VacancyList(props) {
  const { isFetching, statusSearch, jobs, favouritesList } = props;
  return (
    <div>
      {isFetching
        ? <Loader />
        : (
          <div>
            <span className="search__status">{statusSearch}</span>
            <ul>
              {jobs.map((job) => (
                <Vacancy
                  key={job.id}
                  //100% можно написать по другому
                  isFavourite={favouritesList[job.id] ? true : false}
                  {...job}
                />
              ))}
            </ul>
          </div>
        )}
    </div>
  );
}

const mapStateToProps = (state) => ({
  jobs: state.search.jobs,
  statusSearch: state.search.statusSearch,
  isFetching: state.search.isFetching,
  favouritesList: state.favourites.favouritesList,
});

VacancyList.propTypes = {
  jobs: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
  isFetching: PropTypes.bool.isRequired,
  statusSearch: PropTypes.string.isRequired,
  favouritesList: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export default connect(mapStateToProps)(VacancyList);
