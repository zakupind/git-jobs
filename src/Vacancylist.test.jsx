import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { VacancyList } from './Vacancylist';

Enzyme.configure({ adapter: new Adapter() });

describe('Тест листа вакансий', () => {
let component, props;

  beforeEach(() => {
    props = {
      isFetching: false,
      statusSearch: 'Текст статуса',
      jobs: [{
        id: 'aaa7101b-6d63-4887-9e0c-1f2989c3cb36',
        type: 'Full Time',
        url: 'https://jobs.github.com/positions/aaa7101b-6d63-4887-9e0c-1f2989c3cb36',
        created_at: 'Thu Oct 08 20:48:37 UTC 2020',
        company: 'Redbrick',
        company_url: 'https://rdbrck.com/',
        location: 'Victoria, BC, Canada',
        title: 'Senior Application Developer',
        description: '<p>Redbrick is the parent organization',
        how_to_apply: '<p>Please apply to the job link',
        company_logo: 'https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcTZMIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--bfcf8205463e2bed0526f9ae61be3cd93d70e210/rdbrck-wordmark.png',
      }],
      favouritesList: {},
    };

    component = shallow(<VacancyList {...props} />);
  });

  test('render', () => {
    expect(component.length).toBe(1);
  });

  test('render VacancyList', () => {
    expect(component).toMatchSnapshot();
  });
});
