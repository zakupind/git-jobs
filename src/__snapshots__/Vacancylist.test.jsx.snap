// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест листа вакансий render VacancyList 1`] = `
<div>
  <div>
    <span
      className="search__status"
    >
      Текст статуса
    </span>
    <ul>
      <Connect(Vacancy)
        company="Redbrick"
        company_logo="https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcTZMIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--bfcf8205463e2bed0526f9ae61be3cd93d70e210/rdbrck-wordmark.png"
        company_url="https://rdbrck.com/"
        created_at="Thu Oct 08 20:48:37 UTC 2020"
        description="<p>Redbrick is the parent organization"
        how_to_apply="<p>Please apply to the job link"
        id="aaa7101b-6d63-4887-9e0c-1f2989c3cb36"
        isFavourite={false}
        key="aaa7101b-6d63-4887-9e0c-1f2989c3cb36"
        location="Victoria, BC, Canada"
        title="Senior Application Developer"
        type="Full Time"
        url="https://jobs.github.com/positions/aaa7101b-6d63-4887-9e0c-1f2989c3cb36"
      />
    </ul>
  </div>
</div>
`;
